Sources/BranchSDK/Branch.m

```diff
@@ -624,27 +625,20 @@ - (void)initSessionWithLaunchOptions:(NSDictionary *)options
     [self.class addBranchSDKVersionToCrashlyticsReport];
     self.shouldAutomaticallyDeepLink = automaticallyDisplayController;
 
-    // If the SDK is already initialized, this means that initSession was called after other lifecycle calls.
-    if (self.initializationStatus == BNCInitStatusInitialized) {
-        [self initUserSessionAndCallCallback:YES sceneIdentifier:nil urlString:nil];
-        return;
-    }
-
-    // Save data from push notification on app launch
+    // Check for Branch link in a push payload
+    NSString *pushURL = nil;
     #if !TARGET_OS_TV
     if ([options objectForKey:UIApplicationLaunchOptionsRemoteNotificationKey]) {
         id branchUrlFromPush = [options objectForKey:UIApplicationLaunchOptionsRemoteNotificationKey][BRANCH_PUSH_NOTIFICATION_PAYLOAD_KEY];
         if ([branchUrlFromPush isKindOfClass:[NSString class]]) {
             self.preferenceHelper.universalLinkUrl = branchUrlFromPush;
             self.preferenceHelper.referringURL = branchUrlFromPush;
+            pushURL = (NSString *)branchUrlFromPush;
         }
     }
     #endif
 
-    // Handle case where there's no URI scheme or Universal Link.
```

Comment:
This code assumed that there would be a follow up link lifecycle call, but with delayed init that call may have already happened and the associated link saved off. 

Currently researching if this change has side effects.

Reply:
After some testing this appears to be a legacy check that hasn't been required in a long time. It did negatively impact apps are are using just an `AppDelegate` on cold launch with a delayed init. Removing the check fixes the issue.